inc/composer/class-command.php

```diff
@@ -320,18 +335,130 @@ protected function logs( InputInterface $input, OutputInterface $output ) {
 	protected function shell( InputInterface $input, OutputInterface $output ) {
 		$columns = exec( 'tput cols' );
 		$lines = exec( 'tput lines' );
+		$command_prefix = $this->get_base_command_prefix();
 		passthru( sprintf(
-			'cd %s; VOLUME=%s COMPOSE_PROJECT_NAME=%s docker-compose exec -e COLUMNS=%d -e LINES=%d php /bin/bash',
-			'vendor/altis/local-server/docker',
-			escapeshellarg( getcwd() ),
-			$this->get_project_subdomain(),
+			"$command_prefix docker-compose exec -e COLUMNS=%d -e LINES=%d php /bin/bash",
 			$columns,
 			$lines
 		), $return_val );
 
 		return $return_val;
 	}
 
+	protected function db( InputInterface $input, OutputInterface $output ) {
+		$db = $input->getArgument( 'options' )[0] ?? '';
+		$columns = exec( 'tput cols' );
+		$lines = exec( 'tput lines' );
+
+		$base_command_prefix = $this->get_base_command_prefix();
+
+		$base_command = sprintf(
+			"$base_command_prefix docker-compose exec -e COLUMNS=%d -e LINES=%d db",
+			$columns,
+			$lines
+		);
+
+		$return_val = 0;
+
+		switch ( $db ) {
+			case 'info':
+				$connection_data = $this->get_db_connection_data();
+
+				$db_info = <<<EOT
+<info>Root password</info>:  ${connection_data['MYSQL_ROOT_PASSWORD']}
+
+<info>Database</info>:       ${connection_data['MYSQL_DATABASE']}
+<info>User</info>:           ${connection_data['MYSQL_USER']}
+<info>Password</info>:       ${connection_data['MYSQL_PASSWORD']}
+
+<comment>Version</comment>:        ${connection_data['MYSQL_VERSION']}
+<comment>Host</comment>:           ${connection_data['HOST']}
+<comment>Port</comment>:           ${connection_data['PORT']}
+
+EOT;
+				$output->write( $db_info );
+				break;
+			case 'spf':
+				if ( strpos( php_uname(), 'Darwin' ) === false ) {
+					$output->writeln( '<error>This command is only supported on MacOS, use composer server db info to see the database connection details.</error>' );
+					return 1;
+				}
+
+				$connection_data = $this->get_db_connection_data();
+				$spf_file_contents = file_get_contents( dirname( __DIR__, 2 ) . '/templates/sequel.xml' );
+				foreach ( $connection_data as $field_name => $field_value ) {
+					$spf_file_contents = preg_replace( "/(<%=\s)($field_name)(\s%>)/i", $field_value, $spf_file_contents );
+				}
+				$output_file_path = sprintf( '/tmp/%s.spf', $this->get_project_subdomain() );
+				file_put_contents( $output_file_path, $spf_file_contents );
+
+				exec( "open $output_file_path", $null, $return_val );
+				if ( $return_val !== 0 ) {
+					$output->writeln( '<error>You must have Sequel Pro installed to use this command</error>' );
+				}
+
+				break;
+			case null:
+				passthru( "$base_command mysql --database=wordpress --user=root -pwordpress", $return_val );
+				break;
+			default:
+				$output->writeln( "<error>The subcommand $db is not recognized</error>" );
```

Comment:
```suggestion
				$output->writeln( "<error>The subcommand $db is not recognized</error>" );
				return 1;
```
Just adding a non zero exit code

Reply:
I've assigned `1` to `$return_val` instead. Although we already have one early return, it can get confusing if we write more code in that function.

Reply:
good shout, makes sense to me